be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/GetMessagesByIdResponseHandler.scala

```diff
@@ -12,23 +12,32 @@ import ch.epfl.pop.pubsub.graph.{ErrorCodes, GraphMessage, PipelineError, pretty
 import ch.epfl.pop.pubsub.{AskPatternConstants, MessageRegistry, PublishSubscribe}
 
 import scala.annotation.tailrec
+import scala.util.Success
 import scala.concurrent.Await
 
 //This object's job is to handle responses it receives from other servers after sending a heartbeat.
 // When receiving the missing messages, the server's job is to write them on the database.
 object GetMessagesByIdResponseHandler extends AskPatternConstants {
 
-  private final val MAX_RETRY = 5
-  private final val SUCCESS = 0
+  private val MAX_RETRY_PER_MESSAGE = 5
+  private val SUCCESS = 0
 
-  // This function packs each message into a publish before pushing them into the pipeline
+  /** Validate and replay on the system each message received in the get_messages_by_id result
+    *
+    * @param messageRegistry
+    *   The system message registry to use in the validation pipeline
+    * @param system
+    *   Implicit actor system to use the given validator
+    * @return
+    *   Left if some messages couldn't be validated after MAX_RETRY_PER_MESSAGE times, Right for success
+    */
   def responseHandler(messageRegistry: MessageRegistry)(implicit system: ActorSystem): Flow[GraphMessage, GraphMessage, NotUsed] = Flow[GraphMessage].map {
     case Right(JsonRpcResponse(_, Some(resultObject), None, _)) =>
       resultObject.resultMap match {
         case Some(resultMap) =>
-          val receivedResponse: Map[Channel, Set[GraphMessage]] = wrapMsgInPublish(resultMap)
+          val receivedResponse = wrapMsgInPublish(resultMap)
```

Comment:
nitpick: maybe you could pass the number of retries as a parameter to `wrapMsgInPublish` as you used to do with `passThroughPipeline`: otherwise, I feel like it kinds of come from nowhere as `wrapMsgInPublish` as a name doesn't really carry the intent of allocating a number of retry to each message...